src/datasets/summarizations/trend-regression.summarization.ts

```diff
@@ -0,0 +1,139 @@
+    const uWeekday = (p: TimeSeriesPoint) => 1 - uWeekend(p);
+
+    const isWeekend = (p: TimeSeriesPoint) => uWeekend(p) > 0.5;
+    const isWeekday = (p: TimeSeriesPoint) => uWeekday(p) > 0.5;
+
+    const weekdayWeekendRatioPoints = groupPointsByXWeek(points).map(weekPoints => {
+      const week = weekPoints[0].x;
+      const wWeekdays = math.sum(weekPoints.map(uWeekday));
+      const wWeekends = math.sum(weekPoints.map(uWeekend));
+      const sWeekdays = math.sum(weekPoints.map(p => p.y * uWeekday(p)));
+      const sWeekends = math.sum(weekPoints.map(p => p.y * uWeekend(p)));
+
+      if (wWeekdays < 1e-7 || wWeekends < 1e-7) {
+        return { x: week, y: -1 };
+      } else {
+        const avgWeekday = sWeekdays / wWeekdays;
+        const avgWeekend = sWeekends / wWeekends;
+        const weekdayWeekendRatio = avgWeekday / avgWeekend;
+        return { x: week, y: weekdayWeekendRatio };
+      }
+    }).filter(({ y }) => y >= 0);
+
+    const uMostPercentage = trapmfL(0.6, 0.7);
+    const uEqualTraffic = ({ y }) => trapmf(0.8, 0.85, 1.15, 1.2)(y);
+    const weekdayWeekendEqualValidity = sigmaCountQA(weekdayWeekendRatioPoints, uMostPercentage, uEqualTraffic);
+
+
+    const uLinearTrends: [string, MembershipFunction][] = [
+      ['quickly increasing', uQuicklyIncreasingLinearTrend],
+      ['increasing', uIncreasingLinearTrend],
+      ['constant', uConstantLinearTrend],
+      ['decreasing', uDecreasingLinearTrend],
+      ['quickly decreasing', uQuicklyDecreasingLinearTrend],
+    ];
+
+    const summaries: Summary[] = [];
+    const normalizedPoints = normalizePoints(points.map(timeSeriesPointToNumPoint));
+    const weekdayNormalizedPoints = normalizedPoints.filter((_, i) => isWeekday(points[i]));
+    const weekendNormalizedPoints = normalizedPoints.filter((_, i) => isWeekend(points[i]));
+
+    const overallLinearModel = linearRegression(normalizedPoints);
+    const overallLinearTrendValidity = uSmallRegressionError(overallLinearModel.absoluteErrorStd);
+
+    const weekdayLinearModel = linearRegression(weekdayNormalizedPoints);
+    const weekdayLinearTrendValidity = uSmallRegressionError(weekdayLinearModel.absoluteErrorStd);
+
+    const weekendLinearModel = linearRegression(weekendNormalizedPoints);
+    const weekendLinearTrendValidity = uSmallRegressionError(weekendLinearModel.absoluteErrorStd);
+
+    const overallLinearTrendSummariesValidity = Math.max(
+      weekdayWeekendEqualValidity,
+      ...uLinearTrends.map(([_, uLinearTrend]) =>
+        Math.min(
+          overallLinearTrendValidity,
+          uLinearTrend(overallLinearModel.gradientAngleRad),
+          weekdayLinearTrendValidity,
+          uLinearTrend(weekdayLinearModel.gradientAngleRad),
+          weekendLinearTrendValidity,
+          uLinearTrend(weekendLinearModel.gradientAngleRad),
+        )),
+    );
+
+    for (const [linearTrend, uLinearTrend] of uLinearTrends) {

```

Comment:
Can you add a small comment for each of these for blocks? They are all very similar and it takes a bit to figure out what summary each is for.